sssd_test_framework/utils/tools.py

```diff
@@ -599,54 +599,64 @@ def __init__(self, host: MultihostHost) -> None:
         """
         super().__init__(host)
 
-    def passwd(self, name: str | int) -> PasswdEntry | None:
+    def passwd(self, name: str | int, service=None) -> PasswdEntry | None:
         """
         Call ``getent passwd $name``
 
-        :param name: User name or id.
+        :param name: Username or id.
         :type name: str | int
+        :param service: Service used.
+        :type service: str | Defaults to None, optional
         :return: passwd data, None if not found
         :rtype: PasswdEntry | None
         """
-        return self.__exec(PasswdEntry, "passwd", name)
+        return self.__exec(PasswdEntry, "passwd", name, service)
 
-    def group(self, name: str | int) -> GroupEntry | None:
+    def group(self, name: str | int, service=None) -> GroupEntry | None:
         """
         Call ``getent group $name``
 
         :param name: Group name or id.
         :type name: str | int
+        :param service: Service used.
+        :type service: str | Default to None, optional
         :return: group data, None if not found
         :rtype: PasswdEntry | None
         """
-        return self.__exec(GroupEntry, "group", name)
+        return self.__exec(GroupEntry, "group", name, service)
 
     def initgroups(self, name: str) -> InitgroupsEntry:
         """
         Call ``getent initgroups $name``
 
         If ``name`` does not exist, group list is empty. This is standard behavior of ``getent initgroups``
 
-        :param name: User name.
+        :param name: Username.
         :type name: str
         :return: Initgroups data
         :rtype: InitgroupsEntry
         """
         return self.__exec(InitgroupsEntry, "initgroups", name)
 
-    def netgroup(self, name: str) -> NetgroupEntry | None:
+    def netgroup(self, name: str, service=None) -> NetgroupEntry | None:
         """
         Call ``getent netgroup $name``
 
         :param name: Netgroup name.
         :type name: str
+        :param service: Service used.
+        :type service: str | Defaults to None, optional
         :return: Netgroup data, None if not found
         :rtype: NetgroupEntry | None
         """
-        return self.__exec(NetgroupEntry, "netgroup", name)
+        return self.__exec(NetgroupEntry, "netgroup", name, service)
+
+    def __exec(self, cls, cmd: str, name: str | int, service=None) -> Any:
+        if service is not None:
+            service = "-s " + service
+
+        command = self.host.ssh.exec(["getent", cmd, service, name], raise_on_error=False)
```

Comment:
If service is given then the command is `getent passwd None name`. Please test your code.